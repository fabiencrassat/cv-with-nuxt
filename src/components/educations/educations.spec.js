import Educations from './educations';
import testUtils from '~~/lib/tests/testUtils';

describe('educations', () => {
  it('mounts properly', () => {
    expect.hasAssertions();
    testUtils.mountProperly(Educations);
  });

  it('renders properly', () => {
    expect.hasAssertions();
    expect(testUtils.htmlFactory(Educations)).toMatchInlineSnapshot('""');
  });
  it('renders properly with empty object', () => {
    expect.hasAssertions();
    expect(
      testUtils.htmlFactory(Educations, { propsData: { educations: {} } })
    ).toMatchInlineSnapshot('""');
  });
  it('renders properly with props', () => {
    expect.hasAssertions();
    expect(
      testUtils.htmlFactory(Educations, {
        propsData: {
          educations: {
            school1: {
              dates: [2000, 2020],
              id: 'school1',
              label: 'school1',
              description: 'school1 description',
            },
            school2: {
              date: 'jan 2020',
              id: 'school2',
              label: 'school2',
              description: 'school2 description',
            },
            school3: {
              date: 'dec 2000',
              id: 'school3',
              label: 'school3',
              description: 'school3 description',
            },
          },
        },
      })
      // eslint-disable-next-line jest/prefer-inline-snapshots
    ).toMatchSnapshot();
  });
});
