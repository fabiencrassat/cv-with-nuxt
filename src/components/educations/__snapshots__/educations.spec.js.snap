// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`educations renders properly with props 1`] = `
<section class="px-12">
  <ul class="relative">
    <li id="school1" class="relative w-full my-16 ml-8 last:mb-5 first:mt-5"><span class="absolute top-0 left-0"></span>
      <div class="text-lg">
        school1
      </div>
      <div class="italic text-gray-700">
        school1 description
      </div>
      <div class="year"><span class="absolute text-right">2020</span> <span class="absolute text-right">2000</span></div>
    </li>
    <li id="school2" class="relative w-full my-16 ml-8 last:mb-5 first:mt-5"><span class="absolute top-0 left-0"></span>
      <div class="text-lg">
        school2
      </div>
      <div class="italic text-gray-700">
        school2 description
      </div>
      <div class="year"><span class="absolute text-right">jan 2020</span> <span class="absolute text-right"></span></div>
    </li>
    <li id="school3" class="relative w-full my-16 ml-8 last:mb-5 first:mt-5"><span class="absolute top-0 left-0"></span>
      <div class="text-lg">
        school3
      </div>
      <div class="italic text-gray-700">
        school3 description
      </div>
      <div class="year"><span class="absolute text-right">dec 2000</span> <span class="absolute text-right"></span></div>
    </li>
  </ul>
</section>
`;
